diff --git a/gettext-tools/src/write-csharp.c b/gettext-tools/src/write-csharp.c
index 047361a..e3373f6 100644
--- a/gettext-tools/src/write-csharp.c
+++ b/gettext-tools/src/write-csharp.c
@@ -733,7 +733,7 @@ msgdomain_write_csharp (message_list_ty *mlp, const char *canon_encoding,
 
   /* Create the C# file.  */
   register_temp_file (tmpdir, csharp_file_name);
-  csharp_file = fopen_temp (csharp_file_name, "w");
+  csharp_file = fopen (csharp_file_name, "w");
   if (csharp_file == NULL)
     {
       error (0, errno, _("failed to create \"%s\""), csharp_file_name);
@@ -743,7 +743,7 @@ msgdomain_write_csharp (message_list_ty *mlp, const char *canon_encoding,
 
   write_csharp_code (csharp_file, culture_name, class_name, mlp);
 
-  if (fwriteerror_temp (csharp_file))
+  if (fwriteerror (csharp_file))
     {
       error (0, errno, _("error while writing \"%s\" file"), csharp_file_name);
       goto quit3;
